src/Entity/Property/EmailPropertyAwareTrait.php

```diff
@@ -24,18 +24,36 @@ trait EmailPropertyAwareTrait
     protected $email;
 
     /**
-     * @inheritdoc
+    * The original email address of the developer.
+    *
+    * @var string|null
+    */
+    protected $originalEmail;
+
+    /**
+     * {@inheritdoc}
      */
     public function getEmail(): ?string
     {
         return $this->email;
     }
 
     /**
-     * @inheritdoc
+     * {@inheritdoc}
      */
     public function setEmail(string $email): void
     {
         $this->email = $email;
+        if (null === $this->originalEmail) {
+            $this->originalEmail = $email;
+        }
+    }
+
+    /**
+     * The original email address of the developer used when updating user email address.
+     */
+    public function originalEmail(): ?string
```

Comment:
Can this be a private methods?

Reply:
@giteshk  No, this cant be a private/protected method, this is called by `DeveloperController` class from Management API.
This method is only used when updating the user profile and specially when updating users email address.